### DISCUSSION TITLE
Person can be detected twice on one picture

### DISCUSSION THREAD

**Discussion by diabolo86** [immich-app/immich#13311] [Q&A]
### The bug

The picture shows my 2 kids.
The face detection recognize both.
It is possible to set both face to the same person, maybe there should be a warning.
Especially by setting multiple pictures to one person.
The set face is not possible to reset to another person, because it couldn't be reselect.

<img width="1437" alt="2024-10-09 14_46_40" src="https://github.com/user-attachments/assets/0818cebf-be0e-40eb-9084-3afd287aab06">

### The OS that Immich Server is running on

Unraid Docker

### Version of Immich Server

v1.117.0

### Version of Immich Mobile App

v1.117.0 build.162

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
sadly I don't know where to find on unraid
```

### Your .env content

```Shell
sadly I don't know where to find on unraid
```

### Reproduction steps

Review detected people

**Comment by bo0tzz** [comment]
This is working as expected atm

**Comment by mertalev** [comment]
Things like mirrors and collages can lead to the same face showing up multiple times in an image. But I do agree that the re-select UI should let you split the faces up in your example.

**Reply by diabolo86** [reply]
Thank you.
Mirrors and collages make sense to me.
May be a warning would be nice.
A re-select or a selective face recognition of individual images would be great.

**Comment by GSzabados** [comment]
I see the same problem on photos including both of my children. The same person identified twice on the same photo. I have not tried to split it yet, but one person detection list has already 35000 assets, and it is clearly a mix of the two children, plus the photos, where both present and both of them tagged with the same person.

Edit: Even worse, other children from the family are in the mix as well. So under the same person, there are at least 3 people (children), some of them tagged on the same photo together.

**Reply by bquorning** [reply]
I have the exact same problem. I found a separate discussion in https://github.com/immich-app/immich/discussions/14596 as well.

**Comment by pabera** [comment]
I case re-running the face recognition is not working for you (it takes too much time, time spend on naming and merging would be lost), I have come up with a small tool to fix face assignments manually and partially automatically.

* https://github.com/pabera/immich-face-fix
